R/dependecies.R

```diff
@@ -0,0 +1,510 @@
+          }
+        } else if (obj1$is_dependency_package){
+          FALSE
+        } else {
+          TRUE
+        }
+      } else{
+        FALSE
+      }}
+    , logical(1))
+  swap_versions1  <- names1 %in% names2 & swap_version2for1
+  swap_versions2 <- vapply(names1[swap_versions1]
+    , function(n) which(names2 == n)
+    , integer(1))
+  list2[swap_versions2] <- list1[swap_versions1]
+  list2
+}
+
+#' Either use the current lockbox library DESCRIPTION
+#' file or download the accurate remote DESCRIPTION file.
+get_dependencies <- function(package, lock) {
+  locked_package <- package
+  if (locked_package$is_dependency_package &&
+    !is.null(locked_package$latest_version_in_lockbox) &&
+    (is.na(locked_package$version) ||
+      package_version(as.character(locked_package$version)) <
+      package_version(as.character(locked_package$latest_version_in_lockbox)))) {
+      locked_package$version <- locked_package$latest_version_in_lockbox %||% NA
+  }
+  if (!is.na(locked_package$version) && exists_in_lockbox(locked_package)) {
+    dependencies <- dependencies_from_description(locked_package
+      , description_file_for(locked_package$name
+        , gsub("/[^/]+$", "", lockbox_package_path(locked_package))))
+  } else {
+    cat(crayon::blue("."))
+    output <- tryCatch(get_remote_dependencies(package), error = function(e) e)
+    if (is(output, "error")) {
+      message(crayon::red(paste0("Dependencies could not be resolved for package: "
+        , package$name, " version: ", package$version)))
+      dependencies <- list()
+    } else {
+      dependencies <- output
+    }
+  }
+  dependencies <- strip_available_dependencies(dependencies)
+  dependencies <- lapply(dependencies, add_latest_version_in_lockbox)
+  lapply(dependencies, replace_with_lock, lock)
+}
+
+strip_available_dependencies <- function(dependencies) {
+  dependencies <- lapply(dependencies
+    , function(package) {
+      if(package$name %in% pesky_namespaces) NULL
+      else package
+    })
+ dependencies[!vapply(dependencies, is.null, logical(1))]
+}
+
+#' Get the dependencies for a given package
+get_remote_dependencies <- function(package) {
+  UseMethod("get_remote_dependencies")
+}
+
+#' If a package is local we just read from the directory given
+get_remote_dependencies.local <- function(package) {
+  description_name <- file_list$Name[grepl(paste0("^[^/]+"
+    ,"/DESCRIPTION$"), file_list$Name)]
+  dcf <- read.dcf(description_name)
+  dependencies_from_description(package, dcf)
+}
+
+#' For packages on CRAN we will extract to a temporary directory when we
+#' download the accurate remote DESCRIPTION file.
+get_remote_dependencies.CRAN <- function(package) {
+  if (package$is_dependency_package) {
+    package$version <- NA
+  }
+  filepath <- download_package(package)
+  split_fp <- strsplit(filepath, "/")[[1]]
+  dirpath <- paste(split_fp[-length(split_fp)], collapse = "/")
```

Comment:
Also, for what it's worth, use `head(split_fp, -1)` instead of `split_fp[-length(split_fp)]`.